monkey/infection_monkey/network/relay/utils.py

```diff
@@ -26,38 +28,47 @@
 # practical purposes. Revisit this if it's not.
 NUM_FIND_SERVER_WORKERS = 32
 
-IslandAPISearchResults = Dict[str, Optional[IIslandAPIClient]]
+
+@dataclass
+class IslandAPISearchResult:
+    server: SocketAddress
+    client: Optional[IIslandAPIClient]
+
+
+IslandAPISearchResults = List[IslandAPISearchResult]
 
 
 def find_available_island_apis(
-    servers: Iterable[str], island_api_client_factory: AbstractIslandAPIClientFactory
+    servers: Iterable[SocketAddress], island_api_client_factory: AbstractIslandAPIClientFactory
 ) -> IslandAPISearchResults:
     server_list = list(servers)
-    server_iterator = ThreadSafeIterator(server_list.__iter__())
-    server_results: IslandAPISearchResults = {}
+    server_iterator = ThreadSafeIterator(enumerate(server_list.__iter__()))
+    results: Dict[int, IslandAPISearchResult] = {}
```

Comment:
I don't understand what does this int stand for? Index? Then why isn't it just a list?

Reply:
I think this was changed because SocketAddress isn't hashable. I think the right solution it to put this back how it was and just make SocketAddress hashable.